Docs sections: Rename, move Miscellaneous into Guides

bertiewooster opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->
Content issues made more noticeable by changing to Furo Sphinx theme #23159

#### Brief description of what is fixed or changed
Rename three top-level sections to:
- Tutorials
- How-to Guides
- API Reference

Move Miscellaneous topics into Guides, under new heading The SymPy Project

<img width="424" alt="How-to Guides screenshot" src="https://user-images.githubusercontent.com/1335522/160971223-1ddb95c9-9f76-4143-8ba3-d6c1b5f199cd.png">

#### Other comments
Changes are as [I suggested in Furo docs theme pull request](https://github.com/sympy/sympy/pull/23159#issuecomment-1077024911)

#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

bertiewooster committed:
Rename top-level sections to:
Tutorials
How-to Guides
API Reference

bertiewooster committed:
Moved Miscellaneous into Guides,
put under heading
The SymPy project

bertiewooster committed:
Capitalize "Project"

bertiewooster committed:
Use descriptive text for link to PDF version of docs

asmeurer commented:
I missed this PR. I agree with the header changes. For the miscellaneous section, they don't really seem to belong in "guides" from the point of view of the Diataxis organization. But at the same time, I don't know if they're really important enough to list on the main page either. 

I'm a little confused about the current layout. Are these "miscellaneous" pages actually listed anywhere in the main table of contents? I don't see a way to access them, except by going to the [last page](https://docs.sympy.org/dev/modules/stats.html) and clicking "next". 

To be completely honest, the contents of these various pages could probably all be combined together into a single page (but maybe that's an independent thing from the organization).

bertiewooster commented:
> For the miscellaneous section, they don't really seem to belong in "guides" from the point of view of the Diataxis organization. But at the same time, I don't know if they're really important enough to list on the main page either.

Good question; knowing where to put meta information (information about the project) is difficult. I just checked and other similar packages do put meta information under guide:
- NumPy puts for example [license information under User guide](https://numpy.org/doc/stable/user/index.html#user)
- SciPy puts [SciPy Organization information under Developer guide](https://scipy.github.io/devdocs/index.html)

Examining the [Diátaxis Framework](https://diataxis.fr/#diataxis-framework), there is an argument that the best quadrant is Explanation: Understanding-oriented, serve our study (rather than work), theoretical knowledge (rather than practical steps). Explanation already contains [List of active deprecations](https://www.asmeurer.com/sympy-furo-demo/dev/explanation/active-deprecations.html), which seems similar to the project information currently in Miscellaneous.

**tl;dr I think either Guide or Explanation is a reasonable choice, and am happy to go with whichever folks prefer.**

> I'm a little confused about the current layout. Are these "miscellaneous" pages actually listed anywhere in the main table of contents? I don't see a way to access them, except by going to the [last page](https://docs.sympy.org/dev/modules/stats.html) and clicking "next".

The way it is set up now is confusing and sub-optimal. The following is my understanding of how it is currently set up.

Correct, there is no link to the miscellaneous pages from the [documentation home page](https://docs.sympy.org/dev/index.html). The Sphinx directive that dictates the order of pages (top-level sections) on the site is:

```
.. toctree::
   :hidden:

   tutorial/index.rst
   guides/index.rst
   explanation/index.rst
   reference/index.rst
   miscellaneous/index.rst
```

which includes miscellaneous. As `:hidden:` indicates, the `toctree` is hidden, meaning it is not presented as a list of links in the body. Instead, the links to the (four) top-level sections are created explicitly and individually using

`:ref:`SymPy Tutorials <tutorial>``

`:ref:`How-to Guides <guides>``

`:ref:`Explanation <explanation>``

`:ref:`SymPy API Reference <reference>``

As a result, the body of the page, and the table of contents in the left navbar, contains the four explicit top-level sections. However, Sphinx uses the page's `toctree` (even though it's hidden) to determine the order of top-level sections, so when you get to the [last page of the "last" section](https://docs.sympy.org/dev/modules/stats.html), Sphinx sets the next page to the last item in that hidden `toctree`, miscellaneous/index.rst.

> To be completely honest, the contents of these various pages could probably all be combined together into a single page (but maybe that's an independent thing from the organization).

A review of the miscellaneous section seems like a good idea. (One of the pages lists all 1000+ authors, so we might not decide to put all the information on one page.) I concur that's independent of the organization, and potentially a bigger project, so I made a separate issue #23312 so that won't delay fixing this issue.

moorepants commented:
I've mentioned this in older threads before, but I really think all the things you have under "The SymPy Project" is something that belongs on the SymPy website, not in the docs. Moving them as you've done is better than what was before, but if you agree and are more ambitious, I am +1 to moving them to the website.

bertiewooster commented:
@moorepants I have no objections to moving "The SymPy Project" (miscellaneous) pages to https://www.sympy.org--that already contains related material such as Projects using SymPy. If we want to go that route, here's what I suggest:

- For the moment, we leave the miscellaneous pages where they are, so they still exist and can be found using search in the docs, search engines, etc.
- We add to issue #23312 that we plan to move those pages to https://www.sympy.org

I believe the repo for https://www.sympy.org is https://github.com/sympy/sympy.github.com, correct?

moorepants commented:
Sounds good.

> I believe the repo for https://www.sympy.org/ is https://github.com/sympy/sympy.github.com, correct?

Yes.

bertiewooster commented:
Alternatively, we could

1. Go ahead with this pull request as is, so at least the the misc pages are easier to find now (even if on https://docs.sympy.org rather than on the ideal site https://www.sympy.org)
2. Simultaneously
    1. Proceed with #23312
    2. Remove the misc pages from https://docs.sympy.org

I kind of prefer this plan because at least the misc pages would be easy to find immediately.

bertiewooster commented:
With the new theme live in the dev docs, [Miscellaneous](https://docs.sympy.org/dev/miscellaneous/index.html) appears in the sidebar on the [home page](https://docs.sympy.org/dev/index.html). So I suggest we merge this pull request as is, to remove Miscellaneous from the sidebar, then work on #23312 which moves those pages to https://www.sympy.org/.

<img width="226" alt="furo-sidebar" src="https://user-images.githubusercontent.com/1335522/162341692-4b111c7c-afba-45ff-ad39-f561037dbde9.png">

bertiewooster committed:
Fixed spelling of "instructions",
capitalization of "SymPy" (the "P" wasn't capitalized in some places)

bertiewooster committed:
Merge branch 'sympy:master' into docs-sections-names

asmeurer commented:
Looking more closely at the content, this is what I would suggest:

- **SymPy development team**. This is a copy of the AUTHORS file in the repo. It would be difficult to move this to the website because it is generated automatically from the AUTHORS file when the docs are built. I would recommend just replacing this with a link to the AUTHORS file on GitHub, which can be done in a single sentence https://github.com/sympy/sympy/blob/master/AUTHORS.
- **Financial and Infrastructure Support**. This lists all the financial support SymPy has received. It is out of date (it only lists GSoC up to 2020 and doesn't list NumFOCUS, Season of Docs, or CZI). Do we actually need this page? If we do, let's update it. If not, we can move it to the wiki (although most of the information there is already on the wiki), or just delete it. 
- **License**: Just a copy of the license file. This can also be replaced with a link to the file on GitHub, which would just be a single sentence. 
- **Citing SymPy**: This is probably useful to keep somewhere. It's also in the README, but it doesn't hurt to have it in the docs too. This is the first page that shows up when I Google "how to cite sympy".
- **Wiki**: Let's just delete this. The link to the FAQ is bad. The FAQ wiki page isn't actually that great, and if we want to have a real FAQ we should just put it in the docs. Most people don't need to know about the wiki. Most useful content on the wiki ought to just be moved to the docs. 
- **SymPy Papers**. Let's delete this too. The wiki page we can leave there, although it's not being kept up-to-date, as there are currently thousands of papers that mention SymPy.
- **Planet SymPy**. We can keep this somewhere or delete it. It's just a single sentence either way. 
- **SymPy logos**. This is just instructions on how to build the logo. It should go in the documentation guide. 
- **Blogs, News, Magazines**. Another link to a wiki page that hasn't even been updated for 10 years. Let's just delete it. Even if the wiki page is updated, we don't need to cross-link it in the docs (unless we decide to move the wiki page to the docs, but that's a separate discussion).

So to be honest, most of this isn't even that important, and can be either trimmed down or removed. Whatever is left could easily go on a single page in the docs somewhere, or on the website, or the wiki or even the README. These pages are relics from when SymPy was a much younger and smaller project.

bertiewooster committed:
Made changes per Aaron Meurer
Deleted aboutus, outreach, wiki rst files
Moved "SymPy Logos" to documentation style guide (end)
Guides/index: Made the last two links contributing and citing
Replaced citing.rst with citing.md, added link to AUTHORS file

bertiewooster committed:
Merge branch 'docs-sections-names' of github.com:bertiewooster/sympy into docs-sections-names

bertiewooster committed:
Merge branch 'sympy:master' into docs-sections-names

bertiewooster commented:
Here's what I did per @asmeurer's comments, reducing the amount of Miscellaneous pages and keeping the remaining content on the docs site:

- **SymPy development team**. Replaced with a link in the Citing page to the AUTHORS file on GitHub https://github.com/sympy/sympy/blob/master/AUTHORS.
- **Financial and Infrastructure Support**. Deleted.
- **License**: Replaced with a link in the Citing page.
- **Citing SymPy**: Kept, and moved to be the last link on Guides.
- **Wiki**: Deleted.
- **SymPy Papers**. Deleted.
- **Planet SymPy**. Deleted.
- **SymPy logos**. Moved to the end of the documentation guide.
- **Blogs, News, Magazines**. Deleted.

This is ready for review again.

asmeurer commented on doc/src/guides/contributing/documentation-style-guide.rst:
```diff
@@ -417,3 +417,16 @@ Avoid unwelcoming or judgement-based phrases like “That is wrong.” Instead u
+    $ make logo # will be stored in the _build/logo subdirectory
+
+The license of all the logos is the same as SymPy: BSD. See the 
+:ref:`LICENSE file <LICENSE>` in the trunk for more information.
```
This shows how old some of these pages are. "trunk" refers to svn, which SymPy hasn't used since 2007, or maybe even earlier.

bertiewooster replied:
I wondered what trunk meant! I'll remove that term.

asmeurer commented on doc/src/citing.md:
````diff
@@ -30,11 +28,14 @@ A BibTeX entry for LaTeX users is
+```
 
-SymPy is BSD licensed, so you are free to use it whatever you like, be it
+[SymPy is BSD licensed](../../LICENSE), so you are free to use it whatever you like, be it
````
This doesn't work. You'll need to either link to the license file on GitHub, or use `.. literalinclude` like we did with the AUTHORS to copy it into the docs.

bertiewooster replied:
Thanks, I thought that might be dicey. I'll link to the [LICENSE on GitHub](https://raw.githubusercontent.com/sympy/sympy/master/LICENSE).

moorepants commented:
If you want to link to sympy papers that are up-to-date, link to this:

https://www.zotero.org/groups/525293/sympy/library

asmeurer commented:
Maybe that zotero link could be mentioned in the citing SymPy section. It seems appropriate there, and shouldn't be more than a sentence or two.

bertiewooster committed:
Fix link to license file on two pages
Add link to papers citing SymPy on Zotero
Remove reference to "trunk" from svn

bertiewooster committed:
Merge branch 'docs-sections-names' of github.com:bertiewooster/sympy into docs-sections-names

bertiewooster commented:
I made those changes, including linking to Zotero. Also, an existing link to the link to the LICENSE file went to matplotlib's license file, so I fixed that.

Ready for review again.

bertiewooster committed:
Merge branch 'sympy:master' into docs-sections-names

asmeurer committed:
Merge branch 'master' into docs-sections-names

asmeurer commented on doc/src/guides/contributing/documentation-style-guide.rst:
```diff
@@ -417,3 +417,16 @@ Avoid unwelcoming or judgement-based phrases like “That is wrong.” Instead u
 Avoid extraneous phrases like, “we just have to do one more thing.”
+
+SymPy Logos
+^^^^^^^^^^^
```
This should go on the page about building the documentation (build-docs.rst).

bertiewooster replied:
Very good, I moved "SymPy Logos" section from _Documentation Style Guide_ to _Build the Documentation_ > Build the Docs, after Local Installation

asmeurer commented:
Just that small comment, and everything else looks good. 

I resolved the conflicts here, which also enabled the docs preview build.

bertiewooster committed:
Move "SymPy Logos" from
Documentation Style Guide to
Build the Documentation > Build the Docs, after Local Installation

bertiewooster committed:
Merge branch 'docs-sections-names' of github.com:bertiewooster/sympy into docs-sections-names

bertiewooster committed:
Remove trailing space from a line

asmeurer commented:
Thanks for this.

bertiewooster commented:
Great, thanks for merging, @asmeurer. Glad to have the consistent, short, top-level section names:
[Tutorials](https://docs.sympy.org/dev/tutorial/index.html)
[How-to Guides](https://docs.sympy.org/dev/guides/index.html)
[Explanation](https://docs.sympy.org/dev/explanation/index.html)
[API Reference](https://docs.sympy.org/dev/reference/index.html)
